app/models/grade.rb

```diff
@@ -38,6 +38,32 @@ def full_log
     self.grading_output&.sub(/#{go.basename}$/, "details.log")
   end
 
+  def can_compare_orca_tap?
+    return false unless has_orca_output?
+    !(self.grading_output.nil? || orca_output['output'].nil?)
+  end
+
+  def orca_tap_comparison
+    local_tap = TapParser.new(File.read(self.grading_output_path))
+    orca_tap = TapParser.new(orca_output['output'])
+    unless local_tap.test_count == orca_tap.test_count
+      return "Test count does not match; Bottlenose: #{local_tap.test_count} / Orca: #{orca_tap.test_count}"
+    end
+    local_names_to_weights = local_tap.tests.map { |t| [t[:comment], t[:info]['weight']] }.to_h
```

Comment:
This assumes all names/comments are unique, which is not guaranteed.  Instead of mapping comments to weights, map comments to lists of weights, then sort those lists and compare the sorted lists of weights.